### PR TITLE
sub: fix subs being invisible after track reinits during pause

### PR DESCRIPTION
**Pull request by Lypheo**
This fixes the issue where switching subtitles or dis- and reenabling subs during pause causes the subs to disapper until the next frame is rendered (i.e. until playback is resumed or a seek/frame step is issued). I am not confident my solution is ideal (for example I am not sure if there isn’t a place where you could just call update_subtitles at a later point in time when the demuxer is guaranteed to be caught up), but it works and has negligible performance impact.

### CODE REVIEW

File: player/core.h [hunk 1]
```diff
@@ -634,7 +634,7 @@ void reinit_sub_all(struct MPContext *mpctx);
 void uninit_sub(struct MPContext *mpctx, struct track *track);
 void uninit_sub_all(struct MPContext *mpctx);
 void update_osd_msg(struct MPContext *mpctx);
-bool update_subtitles(struct MPContext *mpctx, double video_pts);
+bool update_subtitles(struct MPContext *mpctx, double video_pts, bool force_read_ahead);
```
Comment (thebombzen) [inline player/core.h:637]: Does this change the libmpv ABI?

File: player/sub.c [hunk 2]
```diff
@@ -202,9 +205,28 @@ void reinit_sub(struct MPContext *mpctx, struct track *track)
...
+        // However, since this happens concurrently,
+        // there is no guarantee the demuxer has caught up to
+        // current pos yet at the time of calling update_subtitles.
+        // During regular playback this isn’t an issue since
+        // subtitles are updated often enough anyway, however during
+        // pause, they’re only updated once after the track reinit
+        // so we need to loop the call. We also need to set
+        // force_read_ahead to make sure we wait until the
+        // demuxer has caught up to the current position.
+        if (mpctx->paused)
```
Comment (thebombzen) [inline player/sub.c:224]: Why not `while(!update_subtitles(mpctx, mpctx->playback_pts, mpctx->paused));`

File: player/sub.c [hunk 3]
```diff
@@ -81,7 +81,7 @@ void uninit_sub_all(struct MPContext *mpctx)
 }
 
 static bool update_subtitle(struct MPContext *mpctx, double video_pts,
-                            struct track *track)
+                            struct track *track, bool force_read_ahead)
```
Comment (thebombzen) [inline player/sub.c:84]: Is there any particular reason we need this argument and can't just use `mpctx->paused` as the value of this bool instead?

**Comment by Dudemanguy** [comment]
Superseded by a323dfae426e43451f4d3e08a9a63cb7d1c1ace9

**Comment by Lypheo** [comment]
Dudemanguy singlehandedly saving mpv 🙏

**Pull request closed without merging**